### PR TITLE
fix(version): Prioritize `--preid` over existing prerelease ID

### PR DESCRIPTION
**Pull request by moe971**
in order to switch a prerelease id from alpha version to rc I needed to change condition to prioritize preid insteed of an existing preid

## Description

I juste change order of OR statement to take first preid
 const resolvePrereleaseId = existingPreid => preid || (isPrerelease && existingPreid) || "alpha";

## Motivation and Context
On my continuous integration workflow version my app on alpha prerelease, then on it goes to release candidate, so I try tu use lerna version prerelease --preid rc but because my previous tag was  dam/library-client@0.1.1-alpha.10 lerna 3.0.4 used my existing preid.
The solution was to change order on the test statement.

## How Has This Been Tested?
It has been test on my project, it works 👍 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -741,9 +741,9 @@
       "integrity": "sha512-LAQ1d4OPfSJ/BMbI2DuizmYrrkD9JMaTdi2hQTlI53lQ4kRQPyZQRS4CYQ7O66bnBBnP/oYdRxbk++X0xuFU6A=="
     },
     "JSONStream": {
-      "version": "1.3.3",
-      "resolved": "https://registry.npmjs.org/JSONStream/-/JSONStream-1.3.3.tgz",
-      "integrity": "sha512-3Sp6WZZ/lXl+nTDoGpGWHEpTnnC6X5fnkolYZR6nwIfzbxxvA8utPWe1gCt7i0m9uVGsSz2IS8K8mJ7HmlduMg==",
+      "version": "1.3.4",
```
Comment (evocateur) [inline package-lock.json]: The lockfile should not have changed (`npm ci` should be sufficient). Please revert this file

File: commands/diff/__tests__/__snapshots__/diff-command.test.js.snap [hunk 2]
```diff
@@ -44,10 +44,10 @@ index SHA..SHA 100644
 `;
 
 exports[`DiffCommand should diff packages from the most recent tag 1`] = `
-diff --git a/packages/package-1/package.json b/packages/package-1/package.json
+diff --git c/packages/package-1/package.json w/packages/package-1/package.json
 index SHA..SHA 100644
---- a/packages/package-1/package.json
-+++ b/packages/package-1/package.json
+--- c/packages/package-1/package.json
++++ w/packages/package-1/package.json
```
Comment (evocateur) [inline commands/diff/__tests__/__snapshots__/diff-command.test.js.snap]: I'm not sure why these snapshots needed to be updated. Are you using a non-English localization of git?

Comment (moe971) [reply commands/diff/__tests__/__snapshots__/diff-command.test.js.snap]: I'm on an english localization of git, just my OSX is in French. I revert my package-lock, ran `npm ci` the only diff is the new test. I have the same problems with snapshots.

**Comment by evocateur** [comment]
Okay, well, I can assure you the snapshots do not need updating, and the CI build should pass once you revert the snapshot updates. We probably have to pass some flags to git or something, I don't know why it is trying to change the diff prefixes, it's completely unrelated to your change.

**Comment by moe971** [comment]
As you said after revert everything is green on CI, thanks for your help.

**Comment by evocateur** [comment]
I don't think this solves #1214, as it's actually a separate code path (but certainly accesses similar properties).

**Comment by evocateur** [comment]
I think passing an explicit [`--src-prefix=a/` and `--dst-prefix=b/`](https://git-scm.com/docs/git-diff#git-diff---src-prefixltprefixgt) will solve your local snapshot problems. Can you pull from `master` and execute `npm run ci` to verify for me? (no rush, just not certain how to reproduce on my own machine)

**Comment by moe971** [comment]
I sync my repo, 
My current commit match with 3.0.6 tag, I run `npm i` && `npm run ci` I've got 526 lint error for import/unresolved, did I miss something ?

**Comment by evocateur** [comment]
Unless you're making changes to dependencies, I would run `npm ci` instead of `npm i`. All the lint errors are due to a bad node_modules state.

**Pull request merged**